src/tools/mesh_converter.cpp

```diff
@@ -0,0 +1,684 @@
+    for(int i = 0; i < m_zn.size(); ++i)
+    {
+      m_zn[i] = remap_verts[m_zn[i]];
+    }
+
+    // update zn relation
+    znRel = ZNBuilder().fromSet(&zoneSet).toSet(&nodeSet).indices(
+      typename ZNBuilder::IndicesSetBuilder().size(m_zn.size()).data(&m_zn));
+
+    checkValid();
+  }
+
+  PointType getVertex(PositionType index) const { return coords[index]; }
+
+  /// \brief Returns the segment corresponding to the given index (for TOPO_DIM == 1).
+  template <int D = TOPO_DIM, typename std::enable_if<D == 1, int>::type = 0>
+  SimplexType getSimplex(PositionType index) const
+  {
+    const auto zone = znRel[index];
+    return SimplexType(coords[zone[0]], coords[zone[1]]);
+  }
+
+  /// \brief Returns the triangle corresponding to the given index (for TOPO_DIM == 2).
+  template <int D = TOPO_DIM, typename std::enable_if<D == 2, int>::type = 0>
+  SimplexType getSimplex(PositionType index) const
+  {
+    const auto zone = znRel[index];
+    return SimplexType(coords[zone[0]], coords[zone[1]], coords[zone[2]]);
+  }
+
+  /// \brief Returns the tetrahedron corresponding to the given index (for TOPO_DIM == 3).
+  template <int D = TOPO_DIM, typename std::enable_if<D == 3, int>::type = 0>
+  SimplexType getSimplex(PositionType index) const
+  {
+    const auto zone = znRel[index];
+    return SimplexType(coords[zone[0]], coords[zone[1]], coords[zone[2]], coords[zone[3]]);
+  }
+
+public:
+  NodeSet nodeSet;
+  ZoneSet zoneSet;
+
+  ZoneToNodeRelation znRel;
+  Vertices coords;
+
+private:
+  axom::Array<PositionType> m_zn;
+  VertexArray m_coords;
+};
+
+using TetMesh = SimplicialMesh<3, 3>;  // Volumetric tetrahedral mesh in 3D
+using TriMesh = SimplicialMesh<2, 3>;  // Surface triangle mesh in 3D
+
+/// Loads the pro-e tet mesh and returns a TetMesh instance
+/// Fixes orientations of the tets, but defers fixing degeneracies until the vertices have been welded
+bool loadProe(const std::string& filename, TetMesh& slam_mesh, bool verbose)
+{
+  AXOM_ANNOTATE_SCOPE("load Pro-E file");
+
+  namespace mint = axom::mint;
+
+  SLIC_INFO(axom::fmt::format("Reading pro-e file: '{}' ...", filename));
+
+  /// read pro-e mesh into a mint unstructured mesh
+  auto mint_mesh = std::make_unique<mint::UnstructuredMesh<mint::SINGLE_SHAPE>>(3, axom::mint::TET);
+  {
+    axom::quest::ProEReader reader;
+    reader.setFileName(filename);
+
+    reader.read();
+    SLIC_INFO(axom::fmt::format(axom::utilities::locale(),
+                                "Input mesh has {:L} vertices and {:L} tets",
+                                reader.getNumNodes(),
+                                reader.getNumTets()));
+
+    reader.getMesh(mint_mesh.get());
+  }
+
+  if(mint_mesh == nullptr)
+  {
+    SLIC_ERROR("Mesh is null after reading the proe file.");
+    return false;
+  }
+
+  if(verbose)
+  {
+    std::string vtk_output_file = "proe_tet.vtk";
+    mint::write_vtk(mint_mesh.get(), vtk_output_file);
```

Comment:
Should writing the VTK mesh be inside the verbose check?

Reply:
I was thinking of the VTK output as a debugging aid. 

Should I add a separate command line argument to control whether it gets written?

Reply:
I have no thoughts either way... i just found it slightly odd. Verbose to me usually means more to the screen not an additional output file/type.

Reply:
Thanks for the suggestion -- fixed by adding a new flag